app/src/androidTest/java/org/simple/clinic/user/UserSessionAndroidTest.kt

```diff
@@ -28,81 +25,32 @@ class UserSessionAndroidTest {
   @Inject
   lateinit var userSession: UserSession
 
-  @Inject
-  lateinit var facilityRepository: FacilityRepository
-
   @Inject
   lateinit var appDatabase: AppDatabase
 
   @Inject
   lateinit var testData: TestData
 
   @Inject
-  lateinit var facilityApi: FacilitySyncApi
-
-  private val authenticationRule = AuthenticationRule()
-
-  private val rxErrorsRule = RxErrorsRule()
+  lateinit var facilityRepository: FacilityRepository
 
   @get:Rule
   val ruleChain = RuleChain
-      .outerRule(authenticationRule)
-      .around(rxErrorsRule)!!
+      .outerRule(LocalAuthenticationRule())
+      .around(RxErrorsRule())!!
 
   @Before
   fun setUp() {
     TestClinicApp.appComponent().inject(this)
   }
 
-  @Test
-  fun when_correct_login_params_are_given_then_login_should_happen_and_session_data_should_be_persisted() {
-    val ongoingLoginEntry = userSession.requireLoggedInUser()
-        .map { OngoingLoginEntry(uuid = it.uuid, phoneNumber = it.phoneNumber, pin = testData.qaUserPin()) }
-        .blockingFirst()
-
-    val loginResult = userSession
-        .saveOngoingLoginEntry(ongoingLoginEntry)
-        .andThen(userSession.loginWithOtp(testData.qaUserOtp()))
-        .blockingGet()
-
-    assertThat(loginResult).isInstanceOf(LoginResult.Success::class.java)
-    val (accessToken) = userSession.accessToken()
-    assertThat(accessToken).isNotNull()
-
-    val (loggedInUser) = userSession.loggedInUser().blockingFirst()
-    assertThat(userSession.isUserLoggedIn()).isTrue()
-    assertThat(loggedInUser!!.status).isEqualTo(UserStatus.ApprovedForSyncing)
-    assertThat(loggedInUser.loggedInStatus).isEqualTo(LOGGED_IN)
-  }
-
-  @Test
-  fun when_incorrect_login_params_are_given_then_login_should_fail() {
```

Comment:
This was a useless test because we were testing the implementation of the server here.

Reply:
Nice to see them go!